flight/Modules/Buzzer/buzzer.c

```diff
@@ -0,0 +1,260 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsModules Tau Labs Modules
+ * @{
+ * @addtogroup BuzzerModule Buzzer Module
+ * @{
+ *
+ * @file       buzzer.c
+ * @author     The OpenPilot Team, http://www.openpilot.org Copyright (C) 2010.
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2013-2014
+ * @author     dRonin, http://dronin.org Copyright (C) 2016
+ * @brief      Module to read alarms and turns on/off a buzzer.
+ *
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include "openpilot.h"
+
+#include "flightbuzzersettings.h"
+#include "modulesettings.h"
+#include "flightstatus.h"
+#include "accessorydesired.h"
+#include "pios_thread.h"
+
+// ****************
+// Private constants
+#define STACK_SIZE_BYTES            420  //gcs shows ~40 stack remaining
+#define TASK_PRIORITY               PIOS_THREAD_PRIO_LOW
+#define SAMPLE_PERIOD_MS            200
+// Private types
+
+// Private variables
+static bool module_enabled = false;
+static struct pios_thread *buzzerTaskHandle;
+static int8_t buzzerPin = -1; //Pin for buzzer
+static int8_t accChan = -1; //Accessory Channel
+static bool buzzer_settings_updated;
+static bool wasArmed;
+static bool wasDisarmed;
+static 	FlightBuzzerSettingsData buzzerSettings;
```

Comment:
Many of the variables above should be allocated inside `buzzerTask`, so the module doesn't use heap if it isn't activated. I think only `module_enabled`, `buzzerTaskHandle`, `buzzer_settings_updated` should be allocated above.